cmd/waiter/main.go

```diff
@@ -0,0 +1,113 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package main
+
+import (
+	"log"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+// settings composed by command-line flag values.
+type settings struct {
+	lockFile string        // path to lock file
+	timeout  time.Duration // how long wait for 'done'
+}
+
+const longDesc = `
+# waiter
+
+Idle loop to hold a container (possibly a Kubernetes POD) running while some other
+action happens in the background. It is started by issuing "waiter start" and can
+be stopped with "waiter done", or after timeout.
+
+## Usage
+
+Start the waiting, use --timeout to change how long:
+
+	$ waiter start
+
+You can signal "done" by running:
+
+	$ waiter done
+
+Or, alternatively:
+
+	$ rm -f <lock-file>
+
+## Return-Code
+
+In the case of timeout, the waiter will return error, it only exits gracefully via
+"waiter done", or the removal of the lock-file (before timeout).
+`
+
+var (
+	rootCmd  = newRootCmd()
+	startCmd = newStartCmd()
+	doneCmd  = newDoneCmd()
+)
+
+// defaultTimeout default timeout duration.
+var defaultTimeout = 60 * time.Second
+
+// defaultLockFile default location of the lock-file.
+var defaultLockFile = "/tmp/waiter.lock"
```

Comment:
sanity check - we are talking container level storage here, right?

so we don't have to worry aboutmultilpe containers in a pod employing waiter, right?

before I got here, I was expecting a randomized file name with a predefined / static prefix and generated suffix, where the lock-file option allows the user the override that approach

Reply:
> sanity check - we are talking container level storage here, right?
> 
> so we don't have to worry aboutmultilpe containers in a pod employing waiter, right?

Yes, that's correct.

> before I got here, I was expecting a randomized file name with a predefined / static prefix and generated suffix, where the lock-file option allows the user the override that approach

That's the reason we have the PID stored on the file, so we have something unique (on the POD).

Reply:
ok thanks for confirmation

marking resolved - moving to my next thread with my possible alternative approach